mlir/include/mlir/Dialect/Rock/Tuning/RockTuning.h

```diff
@@ -0,0 +1,43 @@
+//===--------- ConvContext.h - MLIR tuning parameter generation ----------===//
+//
+// Part of the MLIR Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines MLIR base types for tuning
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef MLIR_DIALECT_ROCK_ROCKTUNINGTYPE_H
+#define MLIR_DIALECT_ROCK_ROCKTUNINGTYPE_H
+
+#include "mlir/Dialect/Rock/IR/Rock.h"
+#include "mlir/Dialect/Rock/IR/RockTuningParamAttrInterface.h"
+#include "mlir/IR/BuiltinOps.h"
+
+namespace mlir {
+namespace rock {
+
+// Parameter container holding a parameter and serialized string
+struct ParamEntry {
+  RockTuningParamAttrInterface param;
+  std::string perfString;
+  KernelType primaryOpType;
+};
+
+// Total tuning space
+struct TunableParams {
+  std::vector<RockTuningParamAttrInterface> tuningRange;
+  KernelType primaryOpType;
+  int numHeuristicQuick;
+};
+
+TunableParams *createTunableParams(ModuleOp &mod);
```

Comment:
Question: instead of using pointers, can we use some of MLIR's facilities to store the parameter and the space into the relevant MLIR context? 

(It's entirely fine if the answer is "no", I just want to check on the possibility)

Reply:
I can't find anything useful, there is no enough mlir-c use-case unfortunately.
Using pointer just because the caller needs to provide the ownership of the allocation.

Reply:
It's not about the C API, it's about MLIRContext having the ability to store things